### PR TITLE
docs: query processing

### PR DESCRIPTION
**Pull request by crepererum**
Closes https://github.com/influxdata/idpe/issues/17770 .

### CODE REVIEW

**Comment by crepererum** [comment]
Best to be reviewed as "rich diff"

image

**Comment by alamb** [comment]
> I'll probably extract that readme into that doc and replace the README w/ a pointer. I kinda feel the README is wrong: it describes IOx as a whole, not the specific artifact. Like there's not gRPC code in iox_query.

That makes sense to me - thank you -- maybe as a follow on PR

File: docs/query_processing.md [hunk 1]
```diff
@@ -0,0 +1,293 @@
...
+   - Tables referenced in the query, and their schema and column details
+3. The query engine creates a PhysicalPlan by determining the Chunks that contain data:
+   1. Contacts the ingester for any unpersisted data
+   2. Consults the catalog for the name/location of parquet files
+   3. Prunes (discards at this step) any parquet files
+4. Starts the Physical Plan (Execution  plan) and streams the results back to the client
+
+All objects (both metadata, and data) in steps 2 and 3 are cached except the information from the ingester.
+
+A graphic representation may look like this:
```
Comment (stuartcarnie) [inline docs/query_processing.md:20]:
```suggestion
A graphical representation may look like this:
```

File: docs/query_processing.md [hunk 2]
```diff
@@ -0,0 +1,293 @@
+# InfluxDB IOx -- Query Processing
+
+This document illustrates query processing for SQL and InfluxQL.
+
+**NOTE: There is another query interface called InfluxRPC (implemented in [`iox_query_influxrpc`]) which mostly reflects
+the old TSM storage API. The planning there works significantly different and this is NOT part of this document.**
```
Comment (stuartcarnie) [inline docs/query_processing.md:6]:
```suggestion
> **Note** 
>
> There is another query interface called InfluxRPC (implemented in [`iox_query_influxrpc`]) which mostly reflects the old TSM storage API. The planning there works significantly different and this is NOT part of this document.
```

This syntax renders quite nicely in GitHub:

> **Note** 
>
> There is another query interface called InfluxRPC (implemented in [`iox_query_influxrpc`]) which mostly reflects the old TSM storage API. The planning there works significantly different and this is NOT part of this document.

**Review by stuartcarnie — COMMENTED** [review]
Nice work, @crepererum – thanks for doing this!

File: docs/query_processing.md [hunk 3]
```diff
@@ -0,0 +1,293 @@
...
+    PhysicalPlan[Physical Plan]:::intermediate
+    QueryExec[QueryExecution]:::processor
+    Result[Result]:::systemIO
+
+    Query --> LogicalPlanner --> LogicalPlan --> PhysicalPlanner --> PhysicalPlan --> QueryExec --> Result
+```
+
+
+## Frontend
+We accept queries via [Apache Arrow Flight] with IOx-specific semantics or via better specified [Apache Arrow Flight
```
[the remark this answers is not in the document]
Comment (crepererum) [reply docs/query_processing.md]: done

File: docs/query_processing.md [hunk 4]
```diff
@@ -0,0 +1,293 @@
...
+This document illustrates query processing for SQL and InfluxQL.
+
+**NOTE: There is another query interface called InfluxRPC (implemented in [`iox_query_influxrpc`]) which mostly reflects
+the old TSM storage API. The planning there works significantly different and this is NOT part of this document.**
+
+## Basic Flow
+1. Query arrives from the user (e.g. SQL, InfluxQL)
+2. The query engine creates a `LogicalPlan` by consulting the Catalog to find:
+   - Tables referenced in the query, and their schema and column details
+3. The query engine creates a PhysicalPlan by determining the Chunks that contain data:
```
[the remark this answers is not in the document]
Comment (crepererum) [reply docs/query_processing.md]: done

File: docs/query_processing.md [hunk 5]
```diff
@@ -0,0 +1,293 @@
...
+   1. Contacts the ingester for any unpersisted data
+   2. Consults the catalog for the name/location of parquet files
+   3. Prunes (discards at this step) any parquet files
+4. Starts the Physical Plan (Execution  plan) and streams the results back to the client
+
+All objects (both metadata, and data) in steps 2 and 3 are cached except the information from the ingester.
```
[the remark this answers is not in the document]
Comment (crepererum) [reply docs/query_processing.md:19]: improved wording

**Comment by crepererum** [comment]
> > I'll probably extract that readme into that doc and replace the README w/ a pointer. I kinda feel the README is wrong: it describes IOx as a whole, not the specific artifact. Like there's not gRPC code in iox_query.
> 
> That makes sense to me - thank you -- maybe as a follow on PR

Done, but I mostly rewrote the paragraph because:

- it was outdated (there's no RUB and mutable buffer anymore)
- partly wrong (the planning DOES know the chunk type to use proper DF nodes)

So I rephrased it so that the frontends lower to logical, and the logical->phys. steps knows the chunk types (but the frontend does NOT know the chunk type).

File: iox_query/README.md [hunk 6]
```diff
@@ -1,92 +1,3 @@
...
-     │   SQL Frontend    │  │   gRPC Storage   │ ...  │ Future Frontend  │
-     │                   │  │     Frontend     │      │ (e.g. InfluxQL)  │
-     └───────────────────┘  └──────────────────┘      └──────────────────┘
-```
-
-We are trying to avoid ending up with something like this:
-
-
-```
-                          ┌─────────────────────────────────────────────────┐
```
Comment (alamb) [inline iox_query/README.md:61]: 😢  I really liked this picture :)

**Review by alamb — COMMENTED** [review]
Thanks -- looks good

File: iox_query/README.md (hunk 6 above)
Comment (crepererum) [reply iox_query/README.md:61]: The messy one? I can bring it back if you want :wink:

Comment (alamb) [reply iox_query/README.md:61]: Thank you for https://github.com/influxdata/influxdb_iox/pull/8064 ❤️

**Pull request merged**